examples/3d_examples/RestartExample/restart_example.py

```diff
@@ -0,0 +1,431 @@
+        with_forcing=True,
+        with_free_stream_flow=True,
+        real_t=real_t,
+        num_threads=num_threads,
+        filter_vorticity=True,
+        filter_setting_dict={"order": 1, "type": "multiplicative"},
+    )
+    # ==================FLOW SETUP END=========================
+
+    # ==================FLOW-ROD COMMUNICATOR SETUP START======
+    cosserat_rod_flow_interactor = sps.CosseratRodFlowInteraction(
+        cosserat_rod=flow_past_rod,
+        eul_grid_forcing_field=flow_sim.eul_grid_forcing_field,
+        eul_grid_velocity_field=flow_sim.velocity_field,
+        virtual_boundary_stiffness_coeff=coupling_stiffness,
+        virtual_boundary_damping_coeff=coupling_damping,
+        dx=flow_sim.dx,
+        grid_dim=grid_dim,
+        real_t=real_t,
+        num_threads=num_threads,
+        forcing_grid_cls=sps.CosseratRodSurfaceForcingGrid,
+        surface_grid_density_for_largest_element=surface_grid_density_for_largest_element,
+    )
+    restart_example_simulator.add_forcing_to(flow_past_rod).using(
+        sps.FlowForces,
+        cosserat_rod_flow_interactor,
+    )
+    # ==================FLOW-ROD COMMUNICATOR SETUP END======
+    # =================TIMESTEPPING====================
+    restart_example_simulator.finalize()
+    timestepper = ea.PositionVerlet()
+    do_step, stages_and_updates = ea.extend_stepper_interface(
+        timestepper, restart_example_simulator
+    )
+
+    restart_dir = "restart_data"
+
+    if save_data:
+        # setup flow IO
+        io = spu.EulerianFieldIO(
+            position_field=flow_sim.position_field,
+            eulerian_fields_dict={
+                "vorticity": flow_sim.vorticity_field,
+                "velocity": flow_sim.velocity_field,
+            },
+        )
+        # Initialize sphere IO
+        rod_io = spu.CosseratRodIO(
+            cosserat_rod=flow_past_rod, dim=grid_dim, real_dtype=real_t
+        )
+        # Initialize forcing io
+        forcing_io = spu.IO(dim=grid_dim, real_dtype=real_t)
+        # Add vector field on lagrangian grid
+        forcing_io.add_as_lagrangian_fields_for_io(
+            lagrangian_grid=cosserat_rod_flow_interactor.forcing_grid.position_field,
+            lagrangian_grid_name="cosseratrod",
+            vector_3d=cosserat_rod_flow_interactor.lag_grid_forcing_field,
+        )
+
+    if restart_simulation:
+        # find latest saved data
+        iter_num = []
+        for filename in os.listdir():
+            if "sopht" in filename and "h5" in filename:
+                iter_num.append(int(filename.split("_")[-1].split(".")[0]))
+        latest = max(iter_num)
+        # load sopht data
+        flow_sim.time = io.load(h5_file_name=f"sopht_{latest:04d}.h5")
+        _ = rod_io.load(h5_file_name=f"rod_{latest:04d}.h5")
+        _ = forcing_io.load(h5_file_name=f"forcing_grid_{latest:04d}.h5")
+        _ = ea.load_state(restart_example_simulator, restart_dir, True)
```

Comment:
Can we convert this a function? Where user just imports it from Sopht and inputs file name, and restart is uploaded.